### PR TITLE
[SPARK-50504][SQL] Enable SQL pipe syntax by default

### PR DESCRIPTION
**Pull request by dtenedor**
### What changes were proposed in this pull request?

This PR enables SQL pipe syntax by default by updating the default value of the `spark.sql.operatorPipeSyntaxEnabled` configuration from `Utils.isTesting` to `true`.

### Why are the changes needed?

This feature is ready for general usage. It is fully backwards-compatible, and will cause no breaking changes to existing SQL queries.

### Does this PR introduce _any_ user-facing change?

Yes, see above.

### How was this patch tested?

This PR also cleans up a unit test by removing the assignment of that configuration to true, since that is the default value now.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by dtenedor** [comment]
cc @dongjoon-hyun @gengliangwang @cloud-fan here is the PR to enable the SQL pipe syntax configuration by default.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:5116]: We need test cases to check when the configuration is false.

**Comment by dongjoon-hyun** [comment]
Thank you, @dtenedor . 

+1 for @gengliangwang 's comment because it was practically `true` during all tests.
> We need test cases to check when the configuration is false.

Why don't we make it `false` first to validate it here first, @dtenedor and @gengliangwang ?

**Comment by dongjoon-hyun** [comment]
After checking the failures, we need to wrap all failed test cases by the following guard or similar logic, @dtenedor .
```
withSQLConf(SQLConf.OPERATOR_PIPE_SYNTAX_ENABLED.key -> "true") {
...
}
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
Comment (dtenedor) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:5116]: Sounds good, I added test coverage to exercise the error message with pipe syntax when the config is disabled.

**Review by dtenedor — COMMENTED** [review]
Thanks @dongjoon-hyun @gengliangwang for your reviews!

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
I believe `SQL Pipe` is still immature in the AS-IS branch status.

For instance, without the following PR, even `SQL Pipe` Syntax documentation is wrong because it throws error.
- #48724

https://github.com/apache/spark/blob/bb17665955ad536d8c81605da9a59fb94b6e0162/docs/sql-pipe-syntax.md?plain=1#L212-L213

Although I appreciate @dtenedor 's passion as his main contributor role, we need a full community audit in the `master`-branch status.

So, please don't merge this PR.

**Comment by dtenedor** [comment]
@dongjoon-hyun correct, that query is not working as of today because this subtask is not yet completed [1]. The pending PR [2] should fix it. As mentioned on the review thread for that PR, we probably just need to fix the last remaining test case. If you prefer, we can merge that PR before merging this one so that the documentation reflects the current state, or we can merge this one first. Either way is OK to me, up to you.

[1] https://issues.apache.org/jira/browse/SPARK-49565

[2] https://github.com/apache/spark/pull/48724

**Comment by dongjoon-hyun** [comment]
Let's make `FROM` clause work as it's supposed to, @dtenedor . IIUC, the author of `FROM` clause and [I already asked you](https://github.com/apache/spark/pull/48724#issuecomment-2521124627) to take over that. It's simply revising the test case as you investigated and suggested on that PR.

> If you prefer, we can merge that PR before merging this one so that the documentation reflects the current state, or we can merge this one first. Either way is OK to me, up to you.

**Comment by dtenedor** [comment]
Sure, no problem, I can follow up on that and ping this thread when it's done.

**Comment by dtenedor** [comment]
Greetings @dongjoon-hyun and @gengliangwang, I've synced the latest code from master and this feature should be ready to enable by default now. I've tested it as much as possible, such as with these TPC-DS queries ported to the new syntax [1]. CI is passing, this is ready to review at your convenience.

[1] https://github.com/apache/spark/blob/b2c8b3069ef4f5288a5964af0da6f6b23a769e6b/sql/core/src/test/resources/sql-tests/results/pipe-operators.sql.out#L3449

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
@dongjoon-hyun do you think we can merge it now? I think this feature is ready.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I agree with uou, @cloud-fan . We can test more after enabling it by default. Thank you, @dtenedor , for your hard work and patience too.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by dtenedor** [comment]
@dongjoon-hyun thanks for your reviews!!

**Pull request closed without merging**
